### PR TITLE
Updgrade Adal to version 4.5.0 and Power BI API SDK to 2.3.0

### PR DESCRIPTION
**Pull request by MahirDiab**

### CODE REVIEW

File: App Owns Data/PowerBIEmbedded_AppOwnsData/Web.config [hunk 1]
```diff
@@ -19,7 +19,7 @@
     <add key="reportId" value="" />
 
     <!-- Fill Tenant ID in authorityUrl-->
-    <add key="authorityUrl" value="https://login.microsoftonline.com/Fill_Tenant_ID/oauth2/authorize" />
+    <add key="authorityUrl" value="https://login.microsoftonline.com/Fill_Tenant_ID/" />
```
Comment (eligr) [inline App Owns Data/PowerBIEmbedded_AppOwnsData/Web.config]: please change to "common"

File: App Owns Data/PowerBIEmbedded_AppOwnsData/packages.config [hunk 2]
```diff
@@ -12,11 +12,11 @@
   <package id="Microsoft.AspNet.Web.Optimization" version="1.1.3" targetFramework="net452" />
   <package id="Microsoft.AspNet.WebPages" version="3.2.3" targetFramework="net452" />
   <package id="Microsoft.CodeDom.Providers.DotNetCompilerPlatform" version="2.0.1" targetFramework="net461" />
-  <package id="Microsoft.IdentityModel.Clients.ActiveDirectory" version="3.13.9" targetFramework="net452" />
+  <package id="Microsoft.IdentityModel.Clients.ActiveDirectory" version="4.5.0" targetFramework="net461" />
```
Comment (ali-hamud) [inline App Owns Data/PowerBIEmbedded_AppOwnsData/packages.config]: Please check what is the policy of upgrading version of a Nuget to major version (with breaking change).
Users who will use API SDK will need to upgrade ADAL to a breaking change version of ADAL right?
Question is, does that mean that we also need to increment the Major version?

File: User Owns Data/integrate-report-web-app/PBIWebApp/PBIWebApp.csproj [hunk 3]
```diff
@@ -65,12 +60,14 @@
...
     <Reference Include="System.Data" />
     <Reference Include="System.Data.DataSetExtensions" />
+    <Reference Include="System.IdentityModel" />
     <Reference Include="System.Net" />
     <Reference Include="System.Net.Http" />
     <Reference Include="System.Net.Http.WebRequest" />
     <Reference Include="System.Web.Extensions" />
     <Reference Include="System.Drawing" />
     <Reference Include="System.Web" />
+    <Reference Include="System.Windows.Forms" />
```
Comment (ali-hamud) [inline User Owns Data/integrate-report-web-app/PBIWebApp/PBIWebApp.csproj]: Why do you need this?

File: App Owns Data/PowerBIEmbedded_AppOwnsData/packages.config (hunk 2 above)
Comment (MahirDiab) [reply App Owns Data/PowerBIEmbedded_AppOwnsData/packages.config]: this is not the SDK, its the embed samples

File: User Owns Data/integrate-report-web-app/PBIWebApp/PBIWebApp.csproj (hunk 3 above)
Comment (MahirDiab) [reply User Owns Data/integrate-report-web-app/PBIWebApp/PBIWebApp.csproj]: required by the new ADAL version

**Pull request merged**